examples/advanced/nlp-ner/jobs/bert_ncbi/app/custom/learners/bert_learner.py

```diff
@@ -0,0 +1,355 @@
+        self.model.to(self.device)
+        self.optimizer = AdamW(self.model.parameters(), lr=self.lr)
+
+    def local_train(
+        self,
+        fl_ctx,
+        train_loader,
+        abort_signal: Signal,
+    ):
+        """Typical training logic
+        Total local epochs: self.aggregation_epochs
+        Load data pairs from train_loader
+        Compute loss with self.model
+        Update model
+        """
+        for epoch in range(self.aggregation_epochs):
+            if abort_signal.triggered:
+                return make_reply(ReturnCode.TASK_ABORTED)
+            self.model.train()
+            epoch_len = len(train_loader)
+            self.epoch_global = self.epoch_of_start_time + epoch
+            self.log_info(
+                fl_ctx,
+                f"Local epoch {self.client_id}: {epoch + 1}/{self.aggregation_epochs} (lr={self.lr})",
+            )
+            for i, batch_data in enumerate(train_loader):
+                if abort_signal.triggered:
+                    return make_reply(ReturnCode.TASK_ABORTED)
+
+                mask = batch_data[0]["attention_mask"].squeeze(1).to(self.device)
+                input_id = batch_data[0]["input_ids"].squeeze(1).to(self.device)
+                train_label = batch_data[1].to(self.device)
+
+                # optimize
+                loss, logits = self.model(input_id, mask, train_label)
+
+                self.optimizer.zero_grad()
+                loss.backward()
+                self.optimizer.step()
+
+                current_step = epoch_len * self.epoch_global + i
+                self.writer.add_scalar("train_loss", loss.item(), current_step)
+
+    def local_valid(
+        self,
+        valid_loader,
+        abort_signal: Signal,
+        tb_id_pre=None,
+        record_epoch=None,
+    ):
+        """Typical validation logic
+        Load data pairs from train_loader
+        Compute outputs with model
+        Compute evaluation metric with self.valid_metric
+        Add score to tensorboard record with specified id
+        """
+        self.model.eval()
+        with torch.no_grad():
+            total_acc_val, total_loss_val, val_total = 0, 0, 0
+            val_y_pred, val_y_true = [], []
+            for val_data, val_label in valid_loader:
+                if abort_signal.triggered:
+                    return make_reply(ReturnCode.TASK_ABORTED)
+                val_label = val_label.to(self.device)
+                val_total += val_label.shape[0]
+                mask = val_data["attention_mask"].squeeze(1).to(self.device)
+                input_id = val_data["input_ids"].squeeze(1).to(self.device)
+                # Inference
+                loss, logits = self.model(input_id, mask, val_label)
+                # Add items for metric computation
+                for i in range(logits.shape[0]):
+                    # remove pad tokens
+                    logits_clean = logits[i][val_label[i] != -100]
```

Comment:
Should the `-100` be a configurable option?

Reply:
changed